pkg/apiserver/proxy_test.go

```diff
@@ -36,122 +38,219 @@ import (
 	"k8s.io/kubernetes/pkg/api/rest"
 )
 
-func calculateContentLength(reqBody string, transferEncodings []string) int64 {
-	if len(transferEncodings) == 0 || (len(transferEncodings) == 1 && transferEncodings[0] == "identity") {
-		return int64(len(reqBody))
+func TestProxyRequestContentLengthAndTransferEncoding(t *testing.T) {
+	chunk := func(data []byte) []byte {
+		out := &bytes.Buffer{}
+		chunker := httputil.NewChunkedWriter(out)
+		for _, b := range data {
+			if _, err := chunker.Write([]byte{b}); err != nil {
+				panic(err)
+			}
+		}
+		chunker.Close()
+		out.Write([]byte("\r\n"))
```

Comment:
No, chunkedWriter only handles the chunk body, which ends with "0\r\n". After that comes trailers (we don't have any), and a final CRLF. See http://www.w3.org/Protocols/rfc2616/rfc2616-sec3.html#sec3.6.1

We need to add the "\r\n" ourselves after writing the chunked body (just like https://golang.org/src/net/http/transfer.go#L246 does)